File: samples/design-smells/primitive_roles.js
// VG-SMELL-012 positive: role decided by hardcoded string literals, no enum/policy.
function canAccess(user, req, account) {
  if (user.role === "admin") return grantAll();
  if (req.user.role == "owner") return grantOwner();
  if (account.userType === "manager") return grantSome();
  if (account.permission === "editor") return grantEdit();
  return false;
}

module.exports = { canAccess };

File: samples/design-smells/primitive_roles.py
# VG-SMELL-012 positive (python): three hardcoded role comparisons, no enum/policy.
def can_access(user, req, member):
    if user.role == "admin":
        return grant_all()
    if req.user.role == "owner":
        return grant_owner()
    if member.permission == "editor":
        return grant_edit()
    return False
